features/test-commons/src/main/java/org/codice/ddf/features/test/features/AdminFeatureFile.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.features.test.features;
+
+import static org.ops4j.pax.exam.CoreOptions.maven;
+
+import org.codice.ddf.features.test.DependencyVersionResolver;
+import org.codice.ddf.features.test.Feature;
+import org.codice.ddf.features.test.FeatureFile;
+import org.codice.ddf.features.test.FeatureImpl;
+import org.ops4j.pax.exam.options.UrlProvisionOption;
+import org.ops4j.pax.exam.options.UrlReference;
+
+public class AdminFeatureFile implements FeatureFile {
```

Comment:
❓ These seem like a lot of boilerplate, which may not be terrible because we won't change the features that often but what is this buying me over using the string?

Reply:
My hope was that we would start using these classes to create small scoped test environments. It also leverages some of the boilerplate features file coordinates that don't change and are annoying to specify. I do agree it's a lot of boilerplate, but we also don't create feature files too often. I'm going to try to rethink this approach and see if I can come up with something that allows small scoped integration tests to be written and reduce this boilerplate code as well.